system/mp/mpcomm.cpp

```diff
@@ -1957,12 +1975,25 @@ int CMPConnectThread::run()
 
                 if (_remoteep.isNull() || hostep.isNull())
                 {
-                    // JCSMORE, I think _remoteep really must/should match a IP of this local host
-                    StringBuffer errMsg("MP Connect Thread: invalid remote and/or host ep serialized from ");
                     SocketEndpoint ep;
                     sock->getPeerEndpoint(ep);
-                    ep.getUrlStr(errMsg);
-                    FLLOG(MCoperatorWarning, unknownJob, "%s", errMsg.str());
+                    StringBuffer errMsg;
+                    SocketEndpointV4 zeroTest[2];
+                    memset(zeroTest, 0x0, sizeof(zeroTest));
+                    if (memcmp(id, zeroTest, sizeof(id)))
```

Comment:
Couldn't you just check !(remoteep.isNull() && hostep.isNull()) ? which were set from the deserialialized socket butter equivalent?